lib/oli_web/pow/pow_helpers.ex

```diff
@@ -109,38 +127,58 @@ defmodule OliWeb.Pow.PowHelpers do
   `:invited_user` is assigned to the conn, the invitation token will be passed
   on through the URL query params.
   """
-  def authorization_link(conn, provider, opts \\ []) do
-    query_params = invitation_token_query_params(conn) ++ request_path_query_params(conn)
+  def authorization_link(conn, provider, is_last_provider, opts \\ []) do
+    opts = build_otps(opts, conn, provider)
 
+    button_box = build_button_box(conn, provider)
+
+    link = Link.link(button_box, opts)
+
+    if is_last_provider do
+      link
+    else
+      or_text =
+        Tag.content_tag(:div, "OR", [{:data, [test: [or: "test-data"]]}, class: "text-center"])
+
+      Tag.content_tag(:div, [link, Tag.tag(:br), or_text])
+    end
+  end
+
+  defp build_button_box(conn, provider) do
+    icon = provider_icon(provider)
+    provider_name = provider_name(provider, downcase: true)
+    msg_box = build_msg_box(conn, provider, provider_name)
+    Tag.content_tag(:div, [icon, msg_box], class: provider_name <> "-auth-container")
+  end
+
+  defp build_msg_box(conn, provider, provider_name) do
     msg =
       AuthorizationController.extension_messages(conn).login_with_provider(%{
         conn
         | params: %{"provider" => provider}
       })
 
-    icon = provider_icon(provider)
+    Tag.content_tag(:div, msg, class: provider_name <> "-text-container")
+  end
 
-    path =
-      AuthorizationController.routes(conn).path_for(
-        conn,
-        AuthorizationController,
-        :new,
-        [provider],
-        query_params
-      )
+  defp build_otps(opts, conn, provider) do
+    path = build_authentication_path(conn, provider)
 
     opts = Keyword.merge(opts, to: path)
 
-    opts =
-      Keyword.merge(opts, class: "btn btn-md #{provider_class(provider)} btn-block social-signin")
-
-    provider_name = provider_name(provider, downcase: true)
-
-    msg_box = Tag.content_tag(:div, msg, class: provider_name <> "-text-container")
+    Keyword.merge(opts, class: "btn btn-md #{provider_class(provider)} btn-block social-signin")
```

Comment:
Super nit but you could call `Keyword.merge/2` only once by passing both `to` and `class` as the second function param.